lib/bootstrap_form/form_builder.rb

```diff
@@ -160,8 +160,10 @@ def check_box_with_bootstrap(name, options = {}, checked_value = "1", unchecked_
           html = if options[:skip_label]
             checkbox_html
           else
-            # TODO: Notice we don't seem to pass the ID into the custom control.
-            checkbox_html.concat(label(label_name, label_description, class: label_class))
+            checkbox_html
+              .concat(label(label_name,
+                            label_description,
+                            { class: label_class }.merge(options[:id].present? ? { for: options[:id] } : {})))
```

Comment:
I wonder if it would be better to split out the label options into a separate variable instead of trying to do everything in one line? This is a bit hard to read.